vj4/ui/templates/contest_main.html

```diff
@@ -50,40 +51,83 @@ <h2 class="status_title">{{ _(title) }}</h2>
     {% endif %}
     {% endfor %}
     <div class="section">
+    {% if not tdocs %}
+      {{ nothing.render('There are no contests...') }}
+    {% else %}
       <div class="section__header">
         <h1 class="section__title">{{ _('All Contests') }}</h1>
+        <div class="section__tools">
+          {{ _('Show') }}:
+          <select class="inline select">
+            <option>All</option>
+            <option>Only OI</option>
+            <option>Only ACM/ICPC</option>
+            <!-- TODO: dynamic list -->
```

Comment:
@twd2 We need to generate the list dynamically. Of course we should also implement contest filter.